packages/logging/src/logging-and-telemetry.ts

```diff
@@ -52,10 +52,23 @@ import type {
   MongoshLoggingAndTelemetryArguments,
   MongoshTrackingProperties,
 } from './types';
+import { machineIdSync } from 'node-machine-id';
 
 export function setupLoggingAndTelemetry(
   props: MongoshLoggingAndTelemetryArguments
 ): MongoshLoggingAndTelemetry {
+  if (!props.deviceId) {
+    try {
+      props.deviceId = machineIdSync();
```

Comment:
Let's do our very best to avoid synchronous I/O calls in the mongosh startup path, if we can at all? We already have a mechanism for deferring telemetry in place here, maybe we could use that with the async version of this function?